presto-main/src/main/java/io/prestosql/operator/scalar/ArraySubscriptOperator.java

```diff
@@ -77,7 +77,9 @@ else if (elementType.getJavaType() == Slice.class) {
             methodHandle = METHOD_HANDLE_SLICE;
         }
         else {
-            methodHandle = METHOD_HANDLE_OBJECT;
+            methodHandle = METHOD_HANDLE_OBJECT.asType(
+                    // Force this operator to return the element type instead of Object
```

Comment:
Typically this comment would go before this line (standalone).
But I think it's fine to go without a comment at all here